### DISCUSSION TITLE
Install Fooocus on my Mac pro monterey 12.7. always failed.

### DISCUSSION THREAD

**Discussion by jpyudhi** [lllyasviel/Fooocus#3352] [Q&A] [answered]
Hi all, i wanna ask.. i already follow the instruction on https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#models on the Mac section, i was on the point no. 2 Open the macOS Terminal app and clone this repository with git clone https://github.com/lllyasviel/Fooocus.git.  
what is the meaning of 'clone this repository' ?  i open the terminal app then copy paste `git clone https://github.com/lllyasviel/Fooocus.git`.  the result is like this " Last login: Wed Jul 24 19:34:21 on ttys000
(base) jpyudhi@Mac-Pro ~ % git clone https://github.com/lllyasviel/Fooocus.git.
Cloning into 'Fooocus.git.'...
Username for 'https://github.com': yudhi
Password for 'https://yudhi@github.com': 
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/lllyasviel/Fooocus.git./'
(base) jpyudhi@Mac-Pro ~ % 

please help me.. is there any step that ive done wrong ?  i already install condamini3, pytorch, Xcode..   Thanks

**Comment by mashb1t** [comment]
> remote: Support for password authentication was removed on August 13, 2021.

Please create an SSH key and add it to your Github account for authentication. You can find the instructions here: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account

**Comment by jpyudhi** [comment]
Hi Mashb1t - i already done doing create an SSH key and add it Github account for authentication. then i redo Fooocus instaltion, but still failed at step 2"Open the macOS Terminal app and clone this repository with .."   i attach terminal screenshot here.  can you help solving this.. many thanks
![Screen Shot 2024-07-25 at 00 10 51](https://github.com/user-attachments/assets/580b234d-5f70-4e11-acf1-a574d82dabe7)

**Reply by mashb1t** [reply, marked as answer]
2 ways
1. upload the ssh key and use the ssh url to clone the repository: `git clone git@github.com:lllyasviel/Fooocus.git`
2. create a personal access token and use this instead of password auth: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens

**Reply by jpyudhi** [reply]
Ok seems like its all good, i already doing what you say.. then next i follow all the steps in that Fooocus for Mac, but when i doing step no 7. Launch Fooocus by running python entry_with_update.py.  at the end result in my terminal app written like this (please see attachment) -- i think this is failed ? any suggestion more please..  Thanks
![Screen Shot 2024-07-25 at 12 01 20](https://github.com/user-attachments/assets/8cc5b255-451e-44e4-8954-21eca06b445b)

**Reply by mashb1t** [reply]
Fooocus is only compatible with python 3.10, please make sure you're not using python 3.8. You can check this by executing `python --version` and follow tutorials to upgrade to 3.10.

**Reply by jpyudhi** [reply]
Hi ..soory to bother u again, but i already install phyton 3.10.14 - after i check..  should i re-do the step 7 again ? thanks
![Screen Shot 2024-07-25 at 13 29 17](https://github.com/user-attachments/assets/0f4ff724-abe8-432a-bf65-b4611c0c5597)

**Reply by mashb1t** [reply]
I can only tell that based on your previous screenshots onxxruntime wasn't able to find a matching version due to python version incompatibility
![Screenshot_20240725-083508](https://github.com/user-attachments/assets/d0800bd5-e776-4af5-a0d1-b489d40709a3)

Re-run the package installation and ensure python 3.10 usage.

**Reply by jpyudhi** [reply]
ok. i try that thanks

**Reply by jpyudhi** [reply]
> I can only tell that based on your previous screenshots onxxruntime wasn't able to find a matching version due to python version incompatibility ![Screenshot_20240725-083508](https://private-user-images.githubusercontent.com/9307310/351990009-d0800bd5-e776-4af5-a0d1-b489d40709a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5MDgyNDQsIm5iZiI6MTcyMTkwNzk0NCwicGF0aCI6Ii85MzA3MzEwLzM1MTk5MDAwOS1kMDgwMGJkNS1lNzc2LTRhZjUtYTBkMS1iNDg5ZDQwNzA5YTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjVUMTE0NTQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjE5ZmZmOWI2NzNiOGQxYTEyZDVlNGI0YmIyNmU5ZTg1Mzc0OGY3NTcxNGZmNGZiOWRhNzc2MzhiODZlMWE3OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.BdD6INAPf4L9tnWX4itWzKG4tskxx8ec21GTvSvScwo)
> 
> Re-run the package installation and ensure python 3.10 usage.

hi i already re-run the package ..but seems like still not works.  is it true if foocus did not support mac intel ? https://github.com/lllyasviel/Fooocus/discussions/1198#discussioncomment-7879729

**Reply by mashb1t** [reply]
@jpyudhi this is still the case, Intel Macbooks can not use the MPS backend. You may still be able to generate images using CPU only, but then there's no GPU acceleration and the generation process will take significantly longer.

**Reply by jpyudhi** [reply]
i see. Im givin up then install fooocus on my mac.  i guess i must be change to Bill Gates - PC and start doing Foooocus installation

**Comment by lgwjames** [comment]
you added a "." after git, i used to get this error

**Reply by lgwjames** [reply]
send me a pm, see if i can help as got it installed, but downgraded a lot to 2.31 at mo to get images to work again ( still blank on face swop that i never had before )

**Reply by jpyudhi** [reply]
hi first many thanks for your help.  sorry i didnt know what you mean about this "you added a "." after git".

**Reply by jpyudhi** [reply]
> send me a pm, see if i can help as got it installed, but downgraded a lot to 2.31 at mo to get images to work again ( still blank on face swop that i never had before )

ok i ll send

**Reply by lgwjames** [reply]
> hi first many thanks for your help. sorry i didnt know what you mean about this "you added a "." after git".

git clone https://github.com/lllyasviel/Fooocus.git.

should be git clone https://github.com/lllyasviel/Fooocus.git

**Reply by jpyudhi** [reply]
hi i re-do the phyton miniconda installation and py torch, minocoda3 has success installed, but when install pytorch - it said update failed..
![Screen Shot 2024-07-28 at 19 33 47](https://github.com/user-attachments/assets/8f08e791-179c-4671-a839-5b759182bd6f)
 do u know the solution ?

**Reply by mashb1t** [reply]
Please use python 3.10, Fooocus isn't (yet) compatible with 3.12. You can find more information here: https://github.com/lllyasviel/Fooocus/discussions/3358#discussioncomment-10151784

**Reply by jpyudhi** [reply]
ok i already dwongrade to phytoon 3.10 then when i install foocus - the last step is still error .
![Screen Shot 2024-07-28 at 20 22 57](https://github.com/user-attachments/assets/59e2e7c1-ba2d-4539-a9ee-0d97b4ea76cf)
how to fix this ?

**Reply by mashb1t** [reply]
You seem to be using python < 3.10, we've come full circle.
https://github.com/lllyasviel/Fooocus/discussions/3352#discussioncomment-10144698
https://github.com/lllyasviel/Fooocus/discussions/3352#discussioncomment-10145301

**Reply by jpyudhi** [reply]
I ALREADY FOLLOW ALL THE INSTRUCTION BELOW 6 (SIX TIMES) REPEAT.. FIRST FAILED I RE-DO , UNINSTALL FOOCUS, PHYTON MINICONDA EVERYTHING, START FROM BEGINNING AGAIN.- AND SO ON..UNTILL 6 TIMES.  UNINSTALL PHYTON 3.12. DOWNGRADE TO PHYTON 3.10. - I DO EVERYTHING YOU SAY.. BUT STILL AWALYS FAILEDD. INSTAL THIS FOOCUS ON MAC OS MONTEREY // WHYYYYYYYYY ????????  IM LOSING MY PATIENCE.. SORRY.  YOU SAID USING PHYTON 3.10. BUT WHY THE COMMAND INSTRUCTION AT THE END BELOW WRITE UPDATE PHYTON ? 
-->. Launch Fooocus by running **python entry_with_update.py**. WITH UPDATE PY-- SO THE SYSTEM WILL AWAYS CHANGE PHYTON 3.10 TO UPDATE PHYTON.. MEANWHILE YOU ALWAYS SAID PHYTON DOES ONLY RUNNING IN PHYTON 3.10.  THIS SEEMS LIKE  FULL CIRCLE TO PLAYING USER... 

Mac

Note that the [minimal requirement](https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#minimal-requirement) for different platforms is different.

Mac is not intensively tested. Below is an unofficial guideline for using Mac. You can discuss problems https://github.com/lllyasviel/Fooocus/pull/129.

You can install Fooocus on Apple Mac silicon (M1 or M2) with macOS 'Catalina' or a newer version. Fooocus runs on Apple silicon computers via [PyTorch](https://pytorch.org/get-started/locally/) MPS device acceleration. Mac Silicon computers don't come with a dedicated graphics card, resulting in significantly longer image processing times compared to computers with dedicated graphics cards.

    Install the conda package manager and pytorch nightly. Read the [Accelerated PyTorch training on Mac](https://developer.apple.com/metal/pytorch/) Apple Developer guide for instructions. Make sure pytorch recognizes your MPS device.
    Open the macOS Terminal app and clone this repository with git clone https://github.com/lllyasviel/Fooocus.git.
    Change to the new Fooocus directory, cd Fooocus.
    Create a new conda environment, conda env create -f environment.yaml.
    Activate your new conda environment, conda activate fooocus.
    Install the packages required by Fooocus, pip install -r requirements_versions.txt.
    Launch Fooocus by running python entry_with_update.py. (Some Mac M2 users may need python entry_with_update.py --disable-offload-from-vram to speed up model loading/unloading.) The first time you run Fooocus, it will automatically download the Stable Diffusion SDXL models and will take a significant amount of time, depending on your internet connection.

Use python entry_with_update.py --preset anime or python entry_with_update.py --preset realistic for Fooocus Anime/Realistic Edition.

**Reply by mashb1t** [reply]
@jpyudhi entry_with_update.py has nothing to do with updating python itself, but updating python packages to the fixed version in requirements_versions.txt file and Fooocus to the latest available versions. This is not an issue all users are experiencing but it seems to be an individual issue of your setup as python 3.10 is compatible with onxxruntime 1.18.1 and it works on all my setups (AMD Windows, Macbook M1, Colab and an Intel Windows computer).
Maybe somebody else is able to support you there, and you may also check Google for potential solutions in your case. All the best for future debugging!

**Comment by lgwjames** [comment]
so i had this, i install some of it, then pulled ONLY 2.31 and install went fine

**Reply by jpyudhi** [reply]
> so i had this, i install some of it, then pulled ONLY 2.31 and install went fine

sorry i did not know what you sayin.. what is ONLY 2.31. ? youre explanation is not specific

**Comment by lgwjames** [comment]
so I did the git part. as soon as that worked. 8 did the git checkout v.2.31 then run the install requirements script

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by lgwjames** [comment]
sorted?
